### PR TITLE
Don't overwrite psycopg2 adapters when creating the connection

### PR DESCRIPTION
**Pull request by k4nar**
If an application had registered psycopg2 adapters, the next call to `get_connection` would overwrite them with aiopg's default. As we don't need the adapters registered by aiopg, we can safely tell it not to do that by default.

### CODE REVIEW

File: procrastinate/aiopg_connector.py [hunk 1]
```diff
@@ -16,6 +16,11 @@ def wrap_json(arguments: Dict[str, Any]):
 
 
 def get_connection(dsn="", **kwargs) -> Awaitable[aiopg.Connection]:
+    # tell aiopg not to register adapters for hstore & json by default, as
+    # those are registered at the module level and could overwritte previously
```
Comment (ewjoachim) [inline procrastinate/aiopg_connector.py]:
```suggestion
    # those are registered at the module level and could overwrite previously
```

**Comment by elemoine** [comment]
@k4nar @ewjoachim I am not sure about this change.

`aiopg` does [`register_default_json(self._conn)`](https://github.com/aio-libs/aiopg/blob/7f0debf7a45626aa8ec85c8c714550734c0a8cae/aiopg/connection.py#L556), so the registration of the json typecaster only applies to the connection being created. And with procrastinate creating and using its own connections I don't see where there is a risk of overriding application settings. What do you think?

**Comment by entwanne** [comment]
I don't exactly remember where was the global state, but the issue raised in PeopleCore where we explictly ask to not enable these adapters.

Without these two lines, the adapters were enabled when creating the connection in procrastinate, leading to a failure in PeopleCore.

**Comment by elemoine** [comment]
> Without these two lines, the adapters were enabled when creating the connection in procrastinate,

Without these two lines, if the psycopg2 docs are correct, the json and hstore adapters will be enabled **only** for the procrastinate connection that is being created. So I don't see how this can lead to a failure at the application level.

**Comment by k4nar** [comment]
Maybe it was only a failure in our tests then, because we re-use the same connection (due to a global transaction).

**Comment by elemoine** [comment]
> Maybe it was only a failure in our tests then, because we re-use the same connection (due to a global transaction).

Maybe. I'd be tempted to remove these lines, unless someone comes up with a test-case demonstrating that they can have an impact on the application.

**Comment by k4nar** [comment]
We'll test the PR where you do this with our project and let you know.

**Comment by elemoine** [comment]
@k4nar see https://github.com/peopledoc/procrastinate/pull/133. Thanks.

**Pull request merged**